### PR TITLE
feat(jwt): support lazy public key loading for verification

### PR DESCRIPTION
**Pull request by ab-pm**
I did need to use the callback version of `jwt.verify` and added proper types as well as prevented throwing on functions. @speller's commit came just right :-)

### CODE REVIEW

File: src/postgraphile/withPostGraphileContext.ts [hunk 1]
```diff
@@ -277,8 +277,8 @@ function getSettingsForPgClientTransaction({
   pgSettings,
 }: {
   jwtToken?: string;
-  jwtSecret?: string | Buffer;
-  jwtPublicKey?: string | Buffer;
+  jwtSecret?: jwt.Secret;
```
Comment (benjie) [inline src/postgraphile/withPostGraphileContext.ts]: Should we be doing the same for `jwtSecret`?

Comment (ab-pm) [reply src/postgraphile/withPostGraphileContext.ts]: If you mean `| GetPublicKeyOrSecret`, that doesn't work as [`jwt.sign`](https://github.com/auth0/node-jsonwebtoken#jwtsignpayload-secretorprivatekey-options-callback) does not accept a callback but requires the secret to be available synchronously. That's why I was really happy to see that @speller had just separated the types…

Comment (benjie) [reply src/postgraphile/withPostGraphileContext.ts]: :+1: Cool; always good to have justification. :)

**Comment by ab-pm** [comment]
Thanks for forcing me to write the tests - turns out had not worked by simply relaxing the precondition :-/

**Comment by ab-pm** [comment]
This is weird, `eslint(prettier/prettier)` complains about the trailing commas in my vscode, but `yarn prettier` does actually expect them… Let's make the CI happy.

**Comment by benjie** [comment]
Hmmm; prettier and ESLint should both be pulling from the prettier config file here: https://github.com/graphile/postgraphile/blob/master/prettier.config.js

Trailing commas are expected.

**Comment by ab-pm** [comment]
Is it possible to re-run the CI now that https://github.com/graphile/graphile-engine/pull/546 is merged?

**Comment by benjie** [comment]
Alas we have to ship it first :disappointed: 

Maybe I should merge this into the monorepo after all.

**Comment by benjie** [comment]
Merged in the latest engine.

File: src/postgraphile/withPostGraphileContext.ts [hunk 2]
```diff
@@ -103,7 +103,7 @@ const withDefaultPostGraphileContext: WithPostGraphileContextFn = async (
   // Warning: this is only set if pgForceTransaction is falsy
   const operationType = operation != null ? operation.operation : null;
 
-  const { role: pgRole, localSettings, jwtClaims } = getSettingsForPgClientTransaction({
+  const { role: pgRole, localSettings, jwtClaims } = await getSettingsForPgClientTransaction({
```
[the remark this answers is not in the document]
Comment (ab-pm) [reply src/postgraphile/withPostGraphileContext.ts]: My thoughts exactly :-)

**Pull request merged**